### ISSUE TITLE
Prevent accidental deleting

### ISSUE THREAD

**Issue by fbuys** [fastruby/points#201]
**IMPORTANT: please make sure you ask yourself all intro questions and fill all sections of the template.**

**Before we start...:**

- [x] I checked the documentation and didn't find this feature
- [x] I checked to make sure that this feature has not already been requested

**Branch/Commit:**

main

**Describe the feature:**

Please include a detailed description of the feature you are requesting and any detail on it’s expected behavior.

> **As a user**
> **I add a project that I which to use as some sort of template project**
> **And then I add all the stories with descriptions**
> **I don't want anyone to accidentally delete the template project (since it cannot be recovered)**
> **I don't want anyone to accidentally delete a story in my template project (since it cannot be recovered**

**Problem:**

Please include a detailed description of the problem this feature would solve.

> **As a user**
> **I add a project with stories**
> **So that I can use the project as a template to be cloned for future projects**
> **Now I or anybody on my team can delete the project**
> **And I or anybody on my team can delete stories in the project**
> **There is no way to prevent projects or stories from being deleted**
> **There is no way to recover a project or story after it has been deleted**

**Ideas:**

I see two possible solutions, we can discuss implementing either or both.

1. A permission scheme that prevents unauthorised deletion.
2. Prevent deleting all together, use a different mechanism such as archiving or rejecting
3. Allow soft deleting only so we can recover projects and stories.

**I will abide by the [code of conduct](https://github.com/fastruby/points/blob/main/CODE_OF_CONDUCT.md)**

**Comment by arielj** [comment]
Just a note, archiving a project will prevent any change after this is merged https://github.com/fastruby/points/pull/178.

It also prevents additions though, requiring to un-archive it to add stuff (which may not be bad either, if it's something `locked` then you have to `unlock it` to add stuff too)

**Comment by fbuys** [comment]
Thanks for clarifying @arielj .

It seems a bit like archiving is not really meant to prevent accidental deleting, instead it is a happy coincident.
Perhaps my suggestion is a `nice to have`.

But I do think there is some value in adding a tool that is purposely added to protect a project and stories.
